docs/src/main/java/io/quarkus/docs/generation/CopyJavaExamples.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.docs.generation;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CopyJavaExamples {
+
+    public Path outputPath;
+    public List<Path> paths;
+
+    // ${project.basedir}/../target/asciidoc/generated/examples ${project.basedir}/../integration-tests
+    public static void main(String[] args) throws Exception {
+        if (args.length < 2) {
+            System.err.println("Must specify target output directory (first) followed by at least one source directory");
+            System.exit(1);
+        }
+        CopyJavaExamples javaExamples = new CopyJavaExamples();
+
+        javaExamples.outputPath = Path.of(args[0]);
+        javaExamples.paths = Arrays.stream(args).skip(1)
+                .map(Path::of)
+                .collect(Collectors.toList());
+
+        try {
+            javaExamples.run();
+        } catch (Exception e) {
+            System.err.println("Exception occurred while trying to copy java examples");
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    public void run() throws Exception {
+        if (outputPath != null) {
+            Files.createDirectories(outputPath);
+        }
+
+        for (Path path : paths) {
+            Files.walkFileTree(path, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    if (file.toString().contains("src/main/java/documentation/example/")) {
```

Comment:
ah -thats the "special" sauce...so this is saying that to put docs from integration into docs they *must* be in "src/main/java/documentation/example" folder ?

Two questions:

1) what about kotlin based examples ?
2) why the nested example subdir ? why not just all of documentation ?

Reply:
The package name is one of the things I wanted to ask about actually (as it is fairly arbitrary. I didn't want to overlay the `io.quarkus` namespace so `documentation.example` is where I landed. 

This package may eventually be echoed in quickstarts, as those are often the "final solution". We also tend to use `org.acme` (codegen), but that also feels really arbitrary, and felt less self-explanatory for source being grabbed from the integration-test tree (where documentation.examples is self-explanatory, in my opinion).